### PR TITLE
PUD-922: Reintroduce middle names.

### PR DESCRIPTION
**Pull request by philsegal**
- Handle category for PrevConsMainName.

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt [hunk 1]
```diff
@@ -95,14 +96,23 @@ class RecallNotificationContextFactoryTest {
...
+      Arguments.of(null, "", PreConsMainNameCategory.FIRST_MIDDLE_LAST, "Bobbie Bertie Badger"),
+      Arguments.of(null, "", PreConsMainNameCategory.OTHER, ""),
     )
   }
 
-  @ParameterizedTest(name = "create RecallSummaryContext when hasPreviousConvictionMainName is {0}")
+  @ParameterizedTest(name = "create RecallSummaryContext when hasPreviousConvictionMainName is {0} & category is {2}")
   @MethodSource("prevConsMainNameOptions")
-  fun `create RecallSummaryContext with required details`(hasPrevConsMainName: Boolean, prevConsMainName: String, policeFileName: String) {
+  fun `create RecallSummaryContext with required details`(hasPrevConsMainName: Boolean?, prevConsMainName: String?, prevConsMainNameCategory: PreConsMainNameCategory?, expectedName: String) {
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt:115]: Again I know it was like this before but repeating this whole test for each of these scenarios to me obscures both purposes: can we split this so only a much smaller focused test is repeated for the scenarios?  Ta.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextTest.kt [hunk 2]
```diff
@@ -140,7 +140,7 @@ class RecallNotificationContextTest {
           LocalDate.of(2021, 10, 4),
           RecallLengthDescription(recallLength),
           probationOfficerName,
-          PersonName("Bertie", "Badger"),
+          PersonName("Bertie", lastName = "Badger"),
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextTest.kt:143]: Bertie Badger and Maria Badger - no relation.  Again and again!  So much for variety in test data!  I sense the ✋  of the 🕺  🦡  !

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/DocumentFormattersTest.kt [hunk 3]
```diff
@@ -9,6 +9,6 @@ import uk.gov.justice.digital.hmpps.managerecallsapi.domain.LastName
 class DocumentFormattersTest {
   @Test
   fun `should format PersonName`() {
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/DocumentFormattersTest.kt:11]: Good to add back a test with middleNames != null.

**Review by AndyNewton — CHANGES_REQUESTED** [review]
Few QQs.  Ta.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt (hunk 1 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/recallnotification/RecallNotificationContextFactoryTest.kt:115]: It's being cleared up in the next PR once we remove the boolean

**Review by AndyNewton — APPROVED** [review]
